tlslite/tlsrecordlayer.py

```diff
@@ -912,3 +928,45 @@ def _changeWriteState(self):
 
     def _changeReadState(self):
         self._recordLayer.changeReadState()
+
+
+    def writeHeartbeat(self, type, payload, padding_length):
+        """Start a write operation of heartbeat_request.
+
+        @type payload_length: int
+        @param payload_length: Length of payload.
+
+        @type payload: bytes
+        @param payload: Payload, that we want send in request and
+                        get at response.
+
+        @raise socket.error: If a socket error occurs.
+        """
+        try:
+            if self.closed:
+                raise TLSClosedConnectionError(
+                    "attempt to write to closed connection")
+            heartbeat_request = Heartbeat().create(type, payload,
+                                                   padding_length)
+
+            for result in self._sendMsg(heartbeat_request, \
+                                        randomizeFirstBlock=True):
+                yield result
+        except GeneratorExit:
```

Comment:
with `writeAsync`, we need to call `_shutdown()` on errors but GeneratorExit is one of the exceptions that is normally raised by generators (functions that use `yield`) and it doesn't mean that the connection needs to be closed

that code probably should use `StandardError`, but since the whole tlsrecordlayer and tlsconnection need to be rewritten, I'm leaving them as is and allowing them to collect dept